### ISSUE TITLE
editor: "select" + formatter: "lookup": list does not work for numbers

### ISSUE THREAD

**Issue by KES777** [olifolkerd/tabulator#1648]
**Is your feature request related to a problem? Please describe.**
When we want replace data by readable value: eg. 0 - Sunday, 1 - Monday.
we can use excellent feature: `formatter: "lookup"`
For same purpose we can use `editor: "select"` to hide those internal values.
But internal data not just strings but can be a numbers:

```javascript
var tabledata = [
	{id:1, country_id: 1 },
];
```

Unfortunately keys at lookup list are always strings:
```javascript
var country_list = {
  1: "Germany",
};
var country_list2 = {
  '1': "Germany",
}
```
Here in both cases `1` is always `"1"`

Thus lookup for editor does not work:
![__2](https://user-images.githubusercontent.com/799021/49952538-ca131380-ff04-11e8-85d9-81b68f1e38ad.jpg)

[tabulator source](https://github.com/olifolkerd/tabulator/blob/master/src/js/modules/edit.js#L583)

**Describe the solution you'd like**
I want to suggest to use less strict comparison:

    if(item.value == curentValue){

instead of

    if(item.value === curentValue){

**Describe alternatives you've considered**
We can preprocess data by changing numbers to strings (see "Ukraine" in JSFiddle)

**Additional context**
[JSFiddle to test behavior](https://jsfiddle.net/KES777/oL0xsunv/5/)

Try to edit country with ID:2 and ID:3

PS. Sorry, if I am boring.... =(

**Comment by olifolkerd** [comment]
Hey @KES777 

Thanks for the feedback there. the strict comparison is needed for some users, so i will add an additional editorParam on the next minor release (jan) so the comparison type can be set to better handle usage cases like yours.

Cheers

Oli :)

**Comment by KES777** [comment]
nice, thank you. Would be also nice if that additional editorParam will include description of those use cases.

For example I can not imagine usecase for strict comparison, because keys are always strings...

**Comment by olifolkerd** [comment]
This has been resolved on the 4.2 branch and will be included in modnays release.

Cheers

Oli :)

**Comment by KES777** [comment]
sorry, I just opened new issue #1866 . It seems related to this one.
